src/components/input/Select.tsx

```diff
@@ -545,14 +545,6 @@ function SelectMain<T>({
     selector: '[role="option"]:not([aria-disabled="true"])',
   });
 
-  useLayoutEffect(() => {
-    // Focus toggle button after closing listbox, only if previously focused
-    // element was inside the listbox itself
-    if (!listboxOpen && listboxRef.current!.contains(document.activeElement)) {
-      buttonRef.current!.focus();
-    }
-  }, [buttonRef, listboxOpen]);
```

Comment:
We no longer need to manually return the focus to the button once the listbox is closed, because since the listbox content is removed, the focus is automatically returned to the button, which was the last focused button.